### PR TITLE
[spring] Port two-random-choices to v2

### PR DESCRIPTION
**Pull request by kriskowal**

### CODE REVIEW

**Comment by AlexAPB** [comment]
commandeering

**Comment by AlexAPB** [comment]
@kriskowal ready for review.
1/6 of the `make codecov` tests are failing for some reason ¯\_(ツ)_/¯ 
I've seen this behaviour locally and on other PRs too -- it's flaky.

File: v2/yarpctworandomchoices/list.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+type listOptions struct {
+	capacity int
+	source   rand.Source
+}
+
+var defaultListOptions = listOptions{
+	capacity: 10,
+}
+
+// ListOption customizes the behavior of a fewest pending of two random peers
```
Comment (vyshah) [inline v2/yarpctworandomchoices/list.go]: This wording is pretty confusing

File: v2/yarpctworandomchoices/list.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+// Source is a source of randomness for the peer list.
+func Source(source rand.Source) ListOption {
+	return listOptionFunc(func(options *listOptions) {
+		options.source = source
+	})
+}
+
+// New creates a new fewest pending requests of two random peers peer list.
```
Comment (vyshah) [inline v2/yarpctworandomchoices/list.go]: Same here. Maybe 

> New creates a new PeerList based on the `Two Random Choices` strategy/policy

**Pull request merged**